### ISSUE TITLE
Ignoring triggers breaks savepoint rollbacks in `atomic` blocks, leaving transactions in a failed state

### ISSUE THREAD

**Issue by ralokt** [Opus10/django-pgtrigger#43]
When `transaction.atomic` blocks are nested, django will create a savepoint. If there is an error in the nested block, it will attempt to rollback to the last savepoint. This behavior is well documented [here](https://docs.djangoproject.com/en/2.2/topics/db/transactions/#django.db.transaction.atomic) (see in particular the first admonition re: exception handling) and implemented [here](https://github.com/django/django/blob/2.2/django/db/transaction.py#L227).

More concretely, this breaks use cases like the following:

```python
from django.db import transaction, IntegrityError
import pgconnection

with transaction.atomic():
    with pgconnection.ignore("some_app.SomeModel:write_protect"):
        try:
            with transaction.atomic():
                SomeModel.objects.create(unique_name="some-name")
        except IntegrityError:
                SomeModel.objects.create(unique_name="some-fallback-name-that-will-always-be-unique")
```

Here, instead of the second create succeeding, it will fail because the rollback to the savepoint caused by the inner `atomic` block failed, meaning that the outer transaction is still in a failed state.

The reason this occurs is that [_inject_pgtrigger_ignore](https://github.com/Opus10/django-pgtrigger/blob/master/pgtrigger/core.py#L80) is executed for the rollback command that is sent to the database. This causes `psycopg2.errors.InFailedSqlTransaction: current transaction is aborted, commands ignored until end of transaction block` to be thrown.

A possible fix would be to check `cursor.connection.get_transaction_status()` before attempting to prepend the SQL, checking if it's equal to `psycopg2.extensions.TRANSACTION_STATUS_INERROR`, and aborting if this is the case.

Update: I can confirm that this works if monkey patched in!

**Comment by wesleykendall** [comment]
@ralokt thank you so much for reporting these issues related to transactions and ignoring triggers. I'm going to write a failing test case for this one and get it in the library. Will report back here when it's done.

**Comment by wesleykendall** [comment]
Fixed in https://github.com/Opus10/django-pgtrigger/pull/72, along with a failing test case to illustrate correctness. Will be deployed tonight. FYI I added you as a contributor in the README since both of your issues effectively allowed me to copy/paste the fixes. Thanks again for helping this library become more robust

**Comment by wesleykendall** [comment]
Fix is deployed in version 3.3

**Comment by ralokt** [comment]
Again, you're most welcome, and thank you so much for addressing all of these issues! Also it's an honor to be listed as a contributor - this library has saved me and my team quite a few headaches, super glad that it exists and to be of assistance :)
